ivy/container/gradients.py

```diff
@@ -7,18 +7,184 @@
-        lr,
-        inplace=None,
-        stop_gradients=True,
-        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+    def static_is_variable(
+        x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        exclusive: bool=False,
+        key_chains: Optional[Union[List[str], Dict[str, str]]]=None,
+        to_apply: bool=True,
+        prune_unapplied: bool=False,
+        map_sequences: bool=False
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
+            "is_variable",
+            x,
+            exclusive,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences  
+        )
+        
+    def is_variable(
+        self: ivy.Container,
+        exclusive: bool=False,
+        key_chains: Optional[Union[List[str], Dict[str, str]]]=None,
+        to_apply: bool=True,
+        prune_unapplied: bool=False,
+        map_sequences: bool=False
+        ):
+        return self.static_is_variable(
+            self,
+            exclusive,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences
+            )
+    
+    @staticmethod
+    def static_variable_data(
+        x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        key_chains: Optional[Union[List[str], Dict[str, str]]]=None,
+        to_apply: bool=True,
+        prune_unapplied: bool=False,
+        map_sequences: bool=False
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
+            "variable_data",
+            x,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences  
+        )
+        
+    def variable_data(
+        self: ivy.Container,
+        key_chains: Optional[Union[List[str], Dict[str, str]]]=None,
+        to_apply: bool=True,
+        prune_unapplied: bool=False,
+        map_sequences: bool=False,
+        ):
+        return self.static_variable_data(
+            self,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences
+            )
+        
+    @staticmethod
+    def static_execute_with_gradients(
+        func: Callable,
+        xs: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        retain_grads: Optional[bool]=False,
+        key_chains: Optional[Union[List[str], Dict[str, str]]]=None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
+            "execute_with_gradients",
+            func,
+            xs,
+            retain_grads,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences
+        )
+    
+    def execute_with_gradients(
```

Comment:
we should not include an instance method for `execute_with_gradients`, we should only include instance methods for functions whose **first** argument can be an array. This is a recent change, that will be reflected in the docs shortly.